examples/notebooks/beam-ml/dataframe_api_preprocessing.ipynb

```diff
@@ -0,0 +1,2163 @@
+            "\u001b[0;31mValueError\u001b[0m: No producer for ref_PCollection_PCollection_265"
+          ]
+        }
+      ],
+      "source": [
+        "# Specify the location of source csv file to be processed\n",
+        "source_csv_file = 'gs://apache-beam-samples/nasa_jpl_asteroid/sample_10000.csv'\n",
+        "\n",
+        "# Initialize pipline\n",
+        "p = beam.Pipeline(InteractiveRunner())\n",
+        "\n",
+        "# Create a deferred Beam DataFrame with the contents of our csv file.\n",
+        "beam_df = p | beam.dataframe.io.read_csv('/content/drive/MyDrive/apache beam/dataset/nasa/sample_10000.csv', splittable=True)\n",
+        "\n",
+        "# Drop irrelavant columns/columns with missing values\n",
+        "beam_df = beam_df.drop(['spk_id', 'full_name','diameter', 'albedo', 'diameter_sigma'], axis='columns', inplace=False)\n",
+        "\n",
+        "# Get numerical columns/columns with categorical variables\n",
+        "numerical_cols = beam_df.select_dtypes(include=np.number).columns.tolist()\n",
+        "categorical_cols = list(set(beam_df.columns) - set(numerical_cols))\n",
+        "\n",
+        "# Normalize the numerical variables \n",
+        "beam_df_numerical = beam_df.filter(items=numerical_cols)\n",
+        "beam_df_numerical = (beam_df_numerical - beam_df_numerical.mean())/beam_df_numerical.std()\n",
+        "\n",
+        "\n",
+        "# One-hot encode the categorical variables \n",
+        "beam_df_categorical = beam_df.filter(items=categorical_cols)\n",
+        "# ToDo: one hot-encoding step\n",
+        "\n",
+        "# Merge the normalized variables with the one-hot encoded variables\n",
+        "preprocessed_dataset = beam_df_categorical.merge(beam_df_numerical, left_index = True, right_index = True)\n",
+        "\n",
+        "ib.collect(preprocessed_dataset)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "xZvJTqa3XKI_"
+      },
+      "source": [
+        "# Part II : Process the full dataset with the Distributed Runner\n",
+        "Now that we've showcased how to build and execute the pipeline locally using the Interactive Runner. It's time to execute our pipeline on our full dataset by switching to a distributed runner. For this example, we will exectue our pipeline on [Dataflow](https://cloud.google.com/dataflow/docs/guides/deploying-a-pipeline)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "PROJECT_ID = \"<my-gcp-project>\"\n",
+        "REGION = \"us-west1\"\n",
```

Comment:
Nit: can we default to `us-central1`? Its Carbon Free Energy percentage is higher and might have a more average latency for the east coast.

https://cloud.google.com/sustainability/region-carbon

Reply:
easy fix!